### PR TITLE
 * Add functional tests for sample

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

**Comment by ryanbrandenburg** [comment]
Going to do a bit more cleanup with info I learned on another PR.

File: test/FunctionalTests/AntiforgerySampleTest.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        [Fact]
+        public async Task GetItems_SetsCookie()
+        {
+            // Arrange & Act
+            var response = await Client.GetAsync("http://localhost/api/items");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var content = await response.Content.ReadAsStringAsync();
+            Assert.Contains("lawn", content);
```
Comment (rynowak) [inline test/FunctionalTests/AntiforgerySampleTest.cs]: I don't see anything about cookies.

File: test/FunctionalTests/AntiforgerySampleTest.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        public async Task PostItem_NeedsHeader()
+        {
+            // Arrange
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, "http://localhost/api/items");
+
+            // Act
+            var exception = await Assert.ThrowsAsync<InvalidOperationException>(async () =>
+            {
+                var response = await Client.SendAsync(httpRequestMessage);
+            });
```
Comment (rynowak) [inline test/FunctionalTests/AntiforgerySampleTest.cs]: Just FYI this is going to change to be a 400. What's here is fine.

File: test/FunctionalTests/AntiforgeryTestHelper.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNet.Antiforgery.FunctionalTests
+{
+    public static class AntiforgeryTestHelper
```
Comment (rynowak) [inline test/FunctionalTests/AntiforgeryTestHelper.cs]: Gross

File: Antiforgery.sln [hunk 4]
```diff
@@ -15,6 +15,8 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "samples", "samples", "{D8C4
 EndProject
 Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "AntiforgerySample", "samples\AntiforgerySample\AntiforgerySample.xproj", "{AF9E0784-5EDB-494F-B46C-1A8DA785C49C}"
 EndProject
+Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "FunctionalTests", "test\FunctionalTests\FunctionalTests.xproj", "{8B288810-5A96-4AF5-9836-8BA2D2953203}"
```
Comment (rynowak) [inline Antiforgery.sln]: Follow naming convention

File: test/FunctionalTests/AntiforgeryTestHelper.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            var cookieData = cookie[1];
+
+            return new CookieMetadata()
+            {
+                Key = cookieKey,
+                Value = cookieData
+            };
+        }
+
+        public class CookieMetadata
```
Comment (rynowak) [inline test/FunctionalTests/AntiforgeryTestHelper.cs]: KeyValuePair?

File: test/FunctionalTests/TestFixture.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    public class TestFixture<TStartup> : IDisposable
+    {
+        private readonly TestServer _server;
+
+        public TestFixture()
+        {
+            // RequestLocalizationOptions saves the current culture when constructed, potentially changing response
+            // localization i.e. RequestLocalizationMiddleware behavior. Ensure the saved culture
+            // (DefaultRequestCulture) is consistent regardless of system configuration or personal preferences.
+            using (new CultureReplacer())
```
Comment (rynowak) [inline test/FunctionalTests/TestFixture.cs]: Is this needed?

File: samples/AntiforgerySample/wwwroot/Index.html [hunk 7]
```diff
@@ -38,7 +38,7 @@
             </form>
         </div>
     </div>
-    <script src="bower_components/angular.js"></script>
+    <script src="angular.js"></script>
```
Comment (rynowak) [inline samples/AntiforgerySample/wwwroot/Index.html]: ??? No.

**Comment by ryanbrandenburg** [comment]
:up: :date:

File: .bowerrc [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+  "directory": "wwwroot/lib"
```
Comment (rynowak) [inline .bowerrc]: If you're going to do this then delete all the gulp stuff.

**Comment by rynowak** [comment]
:shipit:

**Pull request merged**